### PR TITLE
Feat(transfer)add transfer dialog

### PR DESCRIPTION
**Pull request by GailMelanie**
# Description

UI for First AAS Transfer

Fixes # (Issue)

## Type of change

Please delete options that are not relevant.

-   [ ] Minor change (non-breaking change, e.g. documentation adaption)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)

# Checklist:

-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] My changes generate no new warnings
-   [ ] I have added tests that prove my fix or my feature works
-   [ ] New and existing tests pass locally with my changes
-   [ ] My changes contain no console logs

### CODE REVIEW

File: src/app/[locale]/viewer/_components/transfer/TargetRespositories.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                                            { submodelRepositories.map((repo, index) => {
+                                                return <MenuItem key={index} value={repo}>{repo}</MenuItem>
+                                            })}
+                                        </TextField>)}
+                                />
+                            </FormControl>
+                        </Box>
+                        <Divider sx={{ mt: 6, mb: 4 }}/>
+                        <DialogActions>
+                            <Button sx={{ mr: 1 }} variant="outlined" onClick={handleSubmit(onSubmit)}>Save & Back to Previous Aas</Button>
```
Comment (XAndreJentzsch) [inline src/app/[locale]/viewer/_components/transfer/TargetRespositories.tsx:121]: The message is added to the i18n file but hardcoded here

Comment (XAndreJentzsch) [reply src/app/[locale]/viewer/_components/transfer/TargetRespositories.tsx:121]: Maybe more a question for the UX review: Should this button be clickable when no Repo is selected? The default selection for the AAS repository is currently empty

File: src/app/[locale]/viewer/_components/transfer/TargetRespositories.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+export type TargetRepositoryFormData = {
+    repository?: string;
+    submodelRepository?: string;
+}
+
+type TargetRepositoryProps = {
+    onSubmitStep: (values: TargetRepositoryFormData, redirectToNew: boolean) => void;
+}
+
+export function TargetRespositories(props: TargetRepositoryProps) {
```
Comment (XAndreJentzsch) [inline src/app/[locale]/viewer/_components/transfer/TargetRespositories.tsx:32]: Typo in Repositories

File: src/app/[locale]/viewer/_components/transfer/TargetRespositories.tsx (hunk 1 above)
Comment (GailMelanie) [reply src/app/[locale]/viewer/_components/transfer/TargetRespositories.tsx]: Was also a feedback from claudette and is now implemented like that

File: src/app/[locale]/viewer/_components/transfer/TransferDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    const [transferDto, setTransferDto] = useState<TransferFormModel>();
+    const [submodelsFromContext,] = useSubmodelState();
+    const [aasFromContext,] = useAasState();
+    const notificationSpawner = useNotificationSpawner();
+
+    useEffect(() => {
+        const transferDtoTemp = { ...transferDto, aas: aasFromContext, submodels: submodelsFromContext }
+
+        setTransferDto(transferDtoTemp)
+    }, [])
```
Comment (JonathanXITASO) [inline src/app/[locale]/viewer/_components/transfer/TransferDialog.tsx]: we may want to execute this on changes to `aas: aasFromContext, submodels: submodelsFromContext`

File: src/app/[locale]/viewer/_components/transfer/TransferDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        } catch (error) {
+            notificationSpawner.spawn({
+                message: error,
+                severity: 'error',
+            });
+        } finally {
+            if (redirectToNew) {
+                // TODO Redirect to new AAS 
+            }
+            notificationSpawner.spawn({
```
Comment (JonathanXITASO) [inline src/app/[locale]/viewer/_components/transfer/TransferDialog.tsx:67]: move this into the try block, else you will log "Error! /n Success!" on an error

File: src/app/[locale]/viewer/_components/transfer/TransferDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        setTransferDto(transferDtoTemp)
+    }, [])
+
+    const handleSubmitRepositoryStep = async (values: TargetRepositoryFormData, redirectToNew: boolean) => {
+        if (!values.repository || !aasFromContext || !values.repository) {
+            return;
+        }
+
+        // This state can be used later to hold the data of multiple steps
+        const transferDtoTemp = {
```
Comment (JonathanXITASO) [inline src/app/[locale]/viewer/_components/transfer/TransferDialog.tsx:48]: this feels a bit duplicated to the effect above.
Maybe merge them together, if possible?

File: src/components/contexts/CurrentAasContext.tsx [hunk 6]
```diff
@@ -23,16 +31,25 @@ export const useAasState = () => {
 export const useRegistryAasState = () => {
     const context = useContext(CurrentAasContext);
     if (!context) {
-        throw new Error('useSubmodelDescriptorState must be used within a CurrentAasContextProvider');
+        throw new Error('registryAasData must be used within a CurrentAasContextProvider');
```
Comment (JonathanXITASO) [inline src/components/contexts/CurrentAasContext.tsx:34]:
```suggestion
        throw new Error('useRegistryAasState must be used within a CurrentAasContextProvider');
```

**Review by JonathanXITASO — APPROVED** [review]

**Review by JonathanXITASO — APPROVED** [review]

**Pull request merged**